src/class/methods.rs

```diff
@@ -133,20 +125,15 @@ pub trait PyMethodsInventory: inventory::Collect {
     fn get_methods(&self) -> &'static [PyMethodDefType];
 }
 
-/// The implementation of this trait defines which methods a Python type has.
-///
-/// For pyclass derived structs this is implemented by collecting all impl blocks through inventory
-pub trait PyMethodsProtocol {
-    /// Returns all methods that are defined for a class
-    fn py_methods() -> Vec<&'static PyMethodDefType>;
-}
+#[doc(hidden)] // Only to be used through the proc macros
+/// For pyclass derived structs, this trait collects method from all impl blocks using inventory.
+pub trait PyMethodsImpl {
```

Comment:
```suggestion
pub trait PyClassMethods {
```

Don't think we have a `PyMethods` type, so `PyMethodsImpl` is slightly confusing imo.

Reply:
Yeah, but it's the implementation detail of `#[pymethods]`.